backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -361,120 +360,56 @@ export class ExperimentService {
-      let twoCharacterId = decisionPoint.twoCharacterId;
-      if (uniqueIdentifiers.indexOf(twoCharacterId) !== -1) {
-        twoCharacterId = this.getUniqueIdentifier(uniqueIdentifiers);
-        logger.info({ message: `Generate new twoCharacterId for Decision Point =>`, details: twoCharacterId });
-        decisionPoint.twoCharacterId = twoCharacterId;
-      }
-      uniqueIdentifiers = [...uniqueIdentifiers, twoCharacterId];
-      return decisionPoint;
-    });
-
-    experiment.partitions = experimentDecisionPoints;
-    experiment.endOn = null;
-    experiment.createdAt = new Date();
-    experiment.state = EXPERIMENT_STATE.INACTIVE;
-    experiment.stateTimeLogs = [];
-
-    let segmentExcludeData = {
-      type: experiment.experimentSegmentExclusion.segment.type,
-      userIds: experiment.experimentSegmentExclusion.segment.individualForSegment.map(user => {
-        return user.userId;
-      }),
-      groups: experiment.experimentSegmentExclusion.segment.groupForSegment.map(group => {
-        return {type: group.type, groupId: group.groupId};
-      }),
-      subSegmentIds: experiment.experimentSegmentExclusion.segment.subSegments.map(subSegment => {
-        return subSegment.id;
-      }),
-      id: experiment.experimentSegmentExclusion.segment.id,
-      name: experiment.experimentSegmentExclusion.segment.name,
-      description: experiment.experimentSegmentExclusion.segment.description,
-      context: experiment.context[0],
-    }
-
-    let segmentIncludeData = {
-      type: experiment.experimentSegmentInclusion.segment.type,
-      userIds: experiment.experimentSegmentInclusion.segment.individualForSegment.map(user => {
-        return user.userId;
-      }),
-      groups: experiment.experimentSegmentInclusion.segment.groupForSegment.map(group => {
-        return {type: group.type, groupId: group.groupId};
-      }),
-      subSegmentIds: experiment.experimentSegmentInclusion.segment.subSegments.map(subSegment => {
-        return subSegment.id;
-      }),
-      id: experiment.experimentSegmentInclusion.segment.id,
-      name: experiment.experimentSegmentInclusion.segment.name,
-      description: experiment.experimentSegmentInclusion.segment.description,
-      context: experiment.context[0]
-    }
-
-    let segmentIncludeDoc: Segment;
-      let segmentExcludeDoc: Segment;
-      try {
-        segmentIncludeDoc = await this.segmentService.upsertSegment(segmentIncludeData, logger);
-      } catch (err) {
-        const error = err as ErrorWithType;
-        error.details = 'Error in updating IncludeSegment in DB';
-        error.type = SERVER_ERROR.QUERY_FAILED;
+  public async importExperiment(experiments: Experiment[], user: User, logger: UpgradeLogger): Promise<Experiment[]> {
+    for (let experiment of experiments) {
+      const duplicateExperiment = await this.experimentRepository.findOne(experiment.id);
+      if (duplicateExperiment && experiment.id !== undefined) {
```

Comment:
`if (duplicateExperiment && experiment.id)` is actually more type-safe, as it covers all falsey values, not just `undefined`. I assume that's the intent.

Reply:
Updated the code